OCPBUGS#10544: Configure the cluster wide proxy for ALBO (#57401)
openshift/openshift-docs: modules/configuring-egress-proxy.adoc

// Module included in the following assemblies:
//
// * networking/aws_load_balancer_operator/configure-egress-proxy-aws-load-balancer-operator.adoc

:_content-type: PROCEDURE
[id="nw-configuring-cluster-wide-proxy_{context}"]

> = Configuring the AWS Load Balancer Operator to trust the Certificate Authority of the cluster-wide proxy

Author: Addressed

> = Configuring the AWS Load Balancer Operator to trust the Certificate Authority of the cluster wide egress proxy

Reviewer: In OCP document I can find two places that mentioned the "egress", like 
```
(1) By default, all cluster egress traffic is proxied, including calls to the cloud provider API for the cloud that hosts your cluster.
(2) The Proxy object is used to manage the cluster-wide egress proxy.
```
In other places we always use the term "cluster-wide proxy", so I think we need to mention egress here, use "cluster-wide proxy" and keep consistent with other chapters.

Author: Addressed

> . Create the config map to contain the CA bundle in the `aws-load-balancer-operator` namespace by running the following command:

Reviewer 2: ```suggestion
. Create the config map to contain the certificate authority (CA) bundle in the `aws-load-balancer-operator` namespace by running the following command:
```

Author: Addressed

> . Create the config map that contains the CA bundle in the `aws-load-balancer-operator` namespace by running the following command:

Reviewer 3: We need to highlight somehow the fact that this configmap doesn't contain the CA bundle just now.
It will be after the second command is run. Maybe we can say something like:
```suggestion
. Create the config map to contain the CA bundle in the `aws-load-balancer-operator` namespace by running the following command:
```

+
[source,terminal]
----
$ oc -n aws-load-balancer-operator create configmap trusted-ca
----

> . To inject the trusted CA into the config map, add the `config.openshift.io/inject-trusted-cabundle=true` label to the config map by running the following command:

Reviewer 3: ```suggestion
. To inject the trusted CA bundle, add the `config.openshift.io/inject-trusted-cabundle=true` label to the config map by running the following command:
```

+
[source,terminal]
----
$ oc -n aws-load-balancer-operator label cm trusted-ca config.openshift.io/inject-trusted-cabundle=true
----

. Update the subscription of the AWS Load Balancer Operator to access the config map in the deployment of the AWS Load Balancer Operator by running the following command:
+
[source,terminal]
----
$ oc -n aws-load-balancer-operator patch subscription aws-load-balancer-operator --type='merge' -p '{"spec":{"config":{"volumes":[{"name":"trusted-ca","configMap":{"name":"trusted-ca"}}],"volumeMounts":[{"name":"trusted-ca","mountPath":"/etc/pki/tls/certs/albo-tls-ca-bundle.crt","subPath":"ca-bundle.crt"}]}}}'
----

> . Wait for the deployment of the AWS Load Balancer Operator to complete. Verify that the CA bundle is added to the `aws-load-balancer-operator-controller-manager` deployment by running the following command:

Reviewer 2: Just a suggestion
```suggestion
. After the deployment of the AWS Load Balancer Operator is completed, verify that the CA bundle is added to the `aws-load-balancer-operator-controller-manager` deployment by running the following command:
```

Author: Addressed

// Module included in the following assemblies:
//
// * networking/aws_load_balancer_operator/configure-egress-proxy-aws-load-balancer-operator.adoc

:_content-type: PROCEDURE
[id="nw-configuring-egress-proxy_{context}"]

> = Configuring the cluster wide egress proxy on the AWS Load Balancer Operator
> 
> After configuring the cluster wide egress proxy on the AWS Load Balancer Operator in the {product-title} cluster, Operator Lifecycle Manager (OLM) automatically updates all the deployments of the Operators with the environment variables such as `HTTP_PROXY`, `HTTPS_PROXY`, and `NO_PROXY`.
> 
> You must trust a custom Certificate Authority (CA) and create the OLM subscription of the AWS Load Balancer Operator.

Reviewer 3: ```suggestion
= Configuring the AWS Load Balancer Operator to trust the cluster wide egress proxy's Certificate Authority (CA)
```

The details about the env variables are common to all the proxy users, they are well described in the egress proxy sections of the OCP docs. We can save some text here.

Author: Addressed

> . Create the config map containing the CA bundle in the `aws-load-balancer-operator` namespace by running the following command:

Reviewer 3: ```suggestion
. Create the config map which will contain the CA bundle in the `aws-load-balancer-operator` namespace by running the following command:
```

Author: Addressed - made it into the present tense

+
[source,terminal]
----
$ oc -n aws-load-balancer-operator create configmap trusted-ca
----

> . Add the `config.openshift.io/inject-trusted-cabundle=true` label to the config map by running the following command:

Reviewer 3: ```suggestion
. To inject the trusted CA into the configmap, add the `config.openshift.io/inject-trusted-cabundle=true` label to it by running the following command:
```

Author: Addressed

+
[source,terminal]
----
$ oc -n aws-load-balancer-operator label cm trusted-ca config.openshift.io/inject-trusted-cabundle=true
----

. Update the subscription of the AWS Load Balancer Operator to access the config map in the deployment of the AWS Load Balancer Operator by running the following command:
+
[source,terminal]
----
$ oc -n aws-load-balancer-operator patch subscription aws-load-balancer-operator --type='merge' -p '{"spec":{"config":{"volumes":[{"name":"trusted-ca","configMap":{"name":"trusted-ca"}}],"volumeMounts":[{"name":"trusted-ca","mountPath":"/etc/pki/tls/certs/albo-tls-ca-bundle.crt","subPath":"ca-bundle.crt"}]}}}'
----

. Wait for a few minutes till the deployment of the AWS Load Balancer Operator is complete. Verify that the CA bundle is added to the `aws-load-balancer-operator-controller-manager` deployment by running the following command:
+
[source,terminal]
----
$ oc -n aws-load-balancer-operator exec deploy/aws-load-balancer-operator-controller-manager -c manager -- ls -l /etc/pki/tls/certs/albo-tls-ca-bundle.crt
----
+
.Example output
[source,terminal]
----
-rw-r--r--. 1 root 1000690000 5875 Jan 11 12:25 /etc/pki/tls/certs/albo-tls-ca-bundle.crt
----

> . Optional: Verify the AWS Load Balancer Operator is restarted every time the configmap changes by running the following command:

Reviewer 3: ```suggestion
. Optional: Restart the AWS Load Balancer Operator's deployment every time the configmap changes by running the following command:
```
